src/Elasticsearch.Net/ConnectionPool/SniffingSortedStickyConnectionPool.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Elasticsearch.Net
+{
+	public class SniffingSortedStickyConnectionPool : IConnectionPool
+	{
+		private readonly ReaderWriterLockSlim _readerWriter = new ReaderWriterLockSlim();
+
+		private readonly Func<IEnumerable<Node>, IEnumerable<Node>> _sortMethod;
+
+		protected IDateTimeProvider DateTimeProvider { get; }
+
+		protected List<Node> InternalNodes { get; set; }
+
+		/// <inheritdoc/>
+		public bool UsingSsl { get; }
+
+		/// <inheritdoc/>
+		public bool SniffedOnStartup { get; set; }
+
+		/// <inheritdoc/>
+		public IReadOnlyCollection<Node> Nodes => this.InternalNodes;
+
+		/// <inheritdoc/>
+		public int MaxRetries => this.InternalNodes.Count - 1;
+
+		/// <inheritdoc/>
+		public bool SupportsReseeding => true;
+
+		/// <inheritdoc/>
+		public bool SupportsPinging => true;
+
+		/// <inheritdoc/>
+		public DateTime LastUpdate { get; protected set; }
+
+		public SniffingSortedStickyConnectionPool(IEnumerable<Uri> uris, Func<IEnumerable<Node>, IEnumerable<Node>> sortMethod, IDateTimeProvider dateTimeProvider = null)
+			: this(uris.Select(uri => new Node(uri)), sortMethod, dateTimeProvider)
+		{
+		}
+
+		public SniffingSortedStickyConnectionPool(IEnumerable<Node> nodes, Func<IEnumerable<Node>, IEnumerable<Node>> sortMethod, IDateTimeProvider dateTimeProvider = null)
```

Comment:
`sortMethod` should have the same signature as `.OrderBy(Descending)()` from LINQ.